File: components/appmgt/org.wso2.carbon.appmgt.api/1.2.1/src/main/java/org/wso2/carbon/appmgt/api/model/Subscription.java
/*
*  Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
*
*  WSO2 Inc. licenses this file to you under the Apache License,
*  Version 2.0 (the "License"); you may not use this file except
*  in compliance with the License.
*  You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing,
* software distributed under the License is distributed on an
* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
* KIND, either express or implied.  See the License for the
* specific language governing permissions and limitations
* under the License.
*/

package org.wso2.carbon.appmgt.api.model;

import java.util.HashSet;
import java.util.Set;

/**
 * Represents a subscription for a web app
 */
public class Subscription {

    public static final String SUBSCRIPTION_TYPE_ENTERPRISE = "ENTERPRISE";
    public static final String SUBSCRIPTION_TYPE_INDIVIDUAL = "INDIVIDUAL";

    private int subscriptionId;
    private int webAppId;
    private int applicationId;
    private String subscriptionType;
    private Set<String> trustedIdps;
    private String subscriptionStatus;
    private String tierId;

    public Subscription() {
        trustedIdps = new HashSet<String>();
    }

    public int getSubscriptionId() {
        return subscriptionId;
    }

    public void setSubscriptionId(int subscriptionId) {
        this.subscriptionId = subscriptionId;
    }

    public int getWebAppId() {
        return webAppId;
    }

    public void setWebAppId(int webAppId) {
        this.webAppId = webAppId;
    }

    public int getApplicationId() {
        return applicationId;
    }

    public void setApplicationId(int applicationId) {
        this.applicationId = applicationId;
    }

    public String getSubscriptionType() {
        return subscriptionType;
    }

    public void setSubscriptionType(String subscriptionType) {
        this.subscriptionType = subscriptionType;
    }

    public Set<String> getTrustedIdps() {
        return trustedIdps;
    }

    public void setTrustedIdps(Set<String> trustedIdps) {
        this.trustedIdps = trustedIdps;
    }

    public void addTrustedIdp(String idpName){
        trustedIdps.add(idpName);
    }

    public void setSubscriptionStatus(String subscriptionStatus) {
        this.subscriptionStatus = subscriptionStatus;
    }

    public String getSubscriptionStatus() {
        return subscriptionStatus;
    }

    public String getTierId() {
        return tierId;
    }

    public void setTierId(String tierId) {
        this.tierId = tierId;
    }

    /**
     * Checks whether the given IDP is one of the trusted IDPs in this subscription.
     * @param authenticatedIDPs Name of the IDP
     * @return true if the given IDP is a trusted IDP, false otherwise.
     */
    public boolean isTrustedIdp(AuthenticatedIDP[] authenticatedIDPs){
        if(trustedIdps == null || trustedIdps.isEmpty()){
            return false;
        }else{
            boolean containsInAuthenticatedIDPs = false;
            for(AuthenticatedIDP authenticatedIDP:authenticatedIDPs){
                containsInAuthenticatedIDPs = trustedIdps.contains(authenticatedIDP.getIdpName());
                if(containsInAuthenticatedIDPs){
                    break;
                }
            }
            return containsInAuthenticatedIDPs;
        }
    }
}

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/site/services/subscription.js
/*
 * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

/*
 Description: This module is used to handle the subscriptions of the user. The supported actions include:
 1. Listing the subscription details of each application owned by the user.
 2. Listing the subscriptions of each application.
 3. Allowing an application to subscribe to an app.
 4. Removing the subscription of an application.
 */
var serviceModule = (function () {

    var log = new Log('Subscriptions service');

    function Subscriber() {
        this.instance = null;
    }

    Subscriber.prototype.init = function (context, session) {
        this.instance = context.module('subscription');
    };

    /*
     Allows an application to subscribe to an application.
     @options.apiName : The name of the API
     @options.apiVersion: The version of the API
     @options.apiTier : The tier of the API
     @options.apiProvider : The provider of the API
     @options.appName: The name of the application
     @options.user:
     */
    Subscriber.prototype.addSubscription = function (options) {

        var appData = {};
        appData['name'] = options.apiName;
        appData['version'] = options.apiVersion;
        appData['provider'] = options.apiProvider;
        var result = this.instance.addAPISubscription(appData, options.subscriptionType, options.apiTier, options.appName,
            options.user, options.enterprises);
        return result;
    };

    Subscriber.prototype.updateVisibility = function (options) {

        var appData = {};
        appData['name'] = options.apiName;
        appData['version'] = options.apiVersion;
        appData['provider'] = options.apiProvider;
        appData['user'] = options.user;
        appData['op_type'] = options.op_type;
        log.info("User : " + options.user);
        var result = this.instance.updateAPPVisibility(appData);
        return result;
    };

    Subscriber.prototype.removeSubscription = function (options) {

        var appData = {};
        appData['name'] = options.apiName;
        appData['version'] = options.apiVersion;
        appData['provider'] = options.apiProvider;
        var result = this.instance.removeAPISubscription(appData, options.apiTier, options.appName, options.user);

        return result;
    };

    Subscriber.prototype.checkSubscriptionWorkflow = function () {

        var result = this.instance.checkSubscriptionWorkflow();
        return result;
    };

    /*
     The function returns all applications that have subscriptions
     options.user: The name of the user whose apps must be returned
     */
    Subscriber.prototype.getAppsWithSubs = function (options) {
        var result = this.instance.getAllSubscriptions(options.user);
        return (result) ? result.applications : [];
    };


    Subscriber.prototype.getSubsForApp = function (options) {
        var result = this.instance.getAPISubscriptionsForApplication(options.user, "DefaultApplication");
        return (result) ? result.subscriptions : [];
    };


    Subscriber.prototype.getSubscription = function (appInfo, appName, subscriptionType, user) {

        var result = this.instance.getAPISubscription(appInfo, appName, subscriptionType, user);
        return result;
    }

    /*
     Returns all available enterprises for the app.
     */
    Subscriber.prototype.getEnterprisesForApplication = function (options) {
        var result = this.instance.getEnterprisesForApplication(options.appName, options.ssoProviderName,
            options.ssoProviderVersion);
        return result;
    }

    return {
        SubscriptionService: Subscriber
    }
})();